runtime/vm/jfr.cpp

```diff
@@ -1120,7 +1120,7 @@ jfrSamplingThreadProc(void *entryArg)
 jboolean
 setJFRRecordingFileName(J9JavaVM *vm, char *fileName)
 {
-	if ((char*)DEFAULT_JFR_FILE_NAME != vm->jfrState.jfrFileName) {
+	if (0 != strncmp(DEFAULT_JFR_FILE_NAME, vm->jfrState.jfrFileName, strlen(DEFAULT_JFR_FILE_NAME))) {
```

Comment:
This will match files which begin with the defined name, not just the file that matches the name exactly, I believe. Is that what you want?